plugins/cc/ccalgo.ml

```diff
@@ -272,15 +272,16 @@ type state =
      mutable rew_depth:int;
      mutable changed:bool;
      by_type: Int.Set.t Typehash.t;
-     mutable gls:Goal.goal Evd.sigma}
+     mutable env:Environ.env;
+     sigma:Evd.evar_map}
 
 let dummy_node =
   {
     clas=Eqto (min_int,{lhs=min_int;rhs=min_int;rule=Congruence});
     cpath=min_int;
     constructors=PacMap.empty;
     vertex=Leaf;
-    term=Symb (mkRel min_int)
+    term=Symb (Evd.empty,EConstr.mkRel min_int)
```

Comment:
You mean using "from_env"? I think it does not matter here (`mkRel min_int` is anyway meaningless), but I can do it if you insist.

Reply:
I am just relaying comments from more knowledgeable people, this is in no way a hard request, but a very very soft "suggestion".

Reply:
These `dummy_node` are not accessed. To enforce this, to the price of an indirection, we could use an option type and define `dummy_node` to be `None`, rather than filling these dummy nodes with an arbitrary choice of meaningless data?

Reply:
I think Evd.empty is fine for dummy values.

Reply:
I am in favor of any change of data structure for which the performance hit is demonstrably acceptable (using benchmarks). Array of option type is fine if the cost is negligible, but binary trees could be tested as well.